validate.js

```diff
@@ -1,22 +1,35 @@
 'use strict';
-var util = require('util'),
-fs = require('fs'),
-jpath = require('json-path'),
-path = require('path');
-
-
-function getValue(obj, key, transFn){
-	if((0 == key.indexOf('/') || 0 == key.indexOf('#/')) && transFn) {
-		return jpath.resolve(obj, key)
-	}
-	return obj[key]
+const util = require('util');
+const fs = require('fs');
+const path = require('path');
+const { JSONPath } = require('jsonpath-plus');
+
+function getValue(obj, key, transFn) {
+  if (key.startsWith('$') && transFn) {
+    let result = JSONPath({
+      path: key,
+      json: obj,
+      wrap: false,
+      resultType: 'value'
+    });
+		if (!Array.isArray(result)) result = [result];
+		return result;
+  }
+  return obj[key]
 }
 
 function hasKey(obj, key, transFn) {
-	if((0 == key.indexOf('/') || 0 == key.indexOf('#/')) && transFn) {
-		return (jpath.resolve(obj, key).length > 0) ? true : false
-	}
-	return key in obj
+  if (key.startsWith('/') && transFn) {
+    let result = JSONPath({
+      path: key,
+      json: obj,
+      wrap: false,
+      resultType: 'all'
+    });
+		if (!Array.isArray(result)) result = [result];
+		return result;
```

Comment:
This should return a boolean here, not an array.

Reply:
Good call